docs/add-a-new-resource.md

```diff
@@ -0,0 +1,71 @@
+
+### Fill out the Resource Schema
+
+In the `internal/service/<service>/<service>.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, choose the correct data type and supply the correct [Schema Behaviors](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors) in order to ensure Terraform knows how to correctly handle the value.
+
+Typically you will add arguments to represent the values that are under control by Terraform, and attributes in order to supply read-only values as references for Terraform. These are distinguished by Schema Behavior.
+
+Attribute names are to specified in `camel_case` as opposed to the AWS API which is `CamelCase`
+
+### Implement CRUD handlers
+These will map planned Terraform state to the AWS API call, or an AWS API response to an applied Terraform state. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
+
+### Write passing Acceptance Tests
+In order to adequately test the resource we will need to write a complete set of Acceptance Tests. You will need an AWS account for this which allows the creation of that resource. See [Writing Acceptance Tests](running-and-writing-acceptance-tests.md) for a detailed guide on how to approach these.
+
+You will need at minimum:
+
+- Basic Test - Tests full lifecycle (CRUD + Import) of a minimal configuration (all required fields, no optional).
+- Disappears Test - Tests what Terraform does if a resource it is tracking can no longer be found.
+- Per Attribute Tests - For each attribute a test should exists which tests that particular attribute in isolation alongside any required fields.
+
+### Create documentation for the resource
+
+Add a file covering the use of the new resource in `website/docs/r/<service>_<name>.md`. You may want to also add examples of the resource in use particularly if its use is complex, or relies on resources in another service. This documentation will appear on the [Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest) when the resource is made available in a provider release. It is fine to link out to AWS Documentation where appropriate, particularly for values which are likely to change.
+
+### Ensure format and lint checks are passing locally
+
+Run `go fmt` to format your code, and install and run all linters to detect and resolve any structural issues with the implementation or documentation.
```

Comment:
```suggestion
Format your code and check linters to detect various issues.
```

We need to work on providing one easy command to run all linters locally. For example, semgrep isn't included in `make lint`. Saying "all linters" is misleading unfortunately.